### PR TITLE
backport: trivial 2024 01 10

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
## Issue being fixed or feature implemented
Trivial backports batch

## What was done?
backports

## How Has This Been Tested?
Building; running tests

## Breaking Changes
None

## Checklist:
  _Go over all the following points, and put an `x` in all the boxes that apply._
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

File: src/validation.h [hunk 1]
```diff
@@ -289,7 +289,8 @@ class CScriptCheck
 
     bool operator()();
 
-    void swap(CScriptCheck &check) {
+    void swap(CScriptCheck& check) noexcept
```
Comment (knst) [inline src/validation.h:292]: I think we should add it to our code in `hdchain.h` also:

```
-void swap(CHDChain& first, CHDChain& second) // nothrow
+void swap(CHDChain& first, CHDChain& second) nothrow
```

File: configure.ac [hunk 2]
```diff
@@ -1498,6 +1498,26 @@ if test x$use_libevent = xyes; then
   CXXFLAGS="$TEMP_CXXFLAGS"
 fi
 
+if test x$use_libevent = xyes; then
```
[the remark this answers is not in the document]
Comment (PastaPastaPasta) [reply configure.ac:1501]: if you look at commit locally the changes are there...

```
commit 07478a126cd0a0898df75c96df6a730794b8f47c
Author: laanwj <126646+laanwj@users.noreply.github.com>
Date:   Thu Jan 13 16:43:12 2022 +0100

    Merge bitcoin/bitcoin#23607: rpc: Pass const char* to evhttp_connection_get_peer for new libevent
    
    c62d763fc313585d79ad833c9d729f6acf2652aa Necessary improvements to make configure work without libevent installed (Perlover)
    091ccc38c2e589b649648cbcc99aca4802f98775 The evhttp_connection_get_peer function from libevent changes the type of the second parameter. Fixing the problem. (Perlover)
    
    Pull request description:
    
      The second parameter of evhttp_connection_get_peer in libevent already has type as `const char **`
      The compilation of bitcoind with the fresh libevent occurs errors
    
      Details: https://github.com/bitcoin/bitcoin/issues/23606
    
    ACKs for top commit:
      laanwj:
        Code review ACK c62d763fc313585d79ad833c9d729f6acf2652aa
      luke-jr:
        tACK c62d763fc313585d79ad833c9d729f6acf2652aa
    
    Tree-SHA512: d1c8062d90bd0d55c582dae2c3a7e5ee1b6c7ca872bf4aa7fe6f45a52ac4a8f59464215759d961f8efde0efbeeade31b08daf9387d7d50d7622baa1c06992d83

diff --git a/configure.ac b/configure.ac
index 1e47963356..c7daaab2fe 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1478,6 +1478,26 @@ if test x$build_bitcoin_cli$build_bitcoind$bitcoin_enable_qt$use_tests$use_bench
   fi
 fi
 
+if test x$use_libevent = xyes; then
+  TEMP_CXXFLAGS="$CXXFLAGS"
+  CXXFLAGS="$CXXFLAGS $EVENT_CFLAGS"
+  AC_MSG_CHECKING([if evhttp_connection_get_peer expects const char**])
+  AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
+      #include <cstdint>
+      #include <event2/http.h>
+    ]], [[
+      evhttp_connection *conn = (evhttp_connection *)1;
+      const char *host;
+      uint16_t port;
+
+      evhttp_connection_get_peer(conn, &host, &port);
+    ]])],
+    [ AC_MSG_RESULT([yes]); AC_DEFINE([HAVE_EVHTTP_CONNECTION_GET_PEER_CONST_CHAR], [1], [Define this symbol if evhttp_connection_get_peer expects const char**]) ],
+    [ AC_MSG_RESULT([no]) ]
+  )
+  CXXFLAGS="$TEMP_CXXFLAGS"
+fi
+
 dnl QR Code encoding library check
 
 if test "x$use_qr" != xno; then
@@ -1787,6 +1807,7 @@ AC_SUBST(HAVE_MM_PREFETCH)
 AC_SUBST(HAVE_STRONG_GETAUXVAL)
 AC_SUBST(HAVE_GMTIME_R)
 AC_SUBST(ANDROID_ARCH)
+AC_SUBST(HAVE_EVHTTP_CONNECTION_GET_PEER_CONST_CHAR)
 AC_CONFIG_FILES([Makefile src/Makefile doc/man/Makefile share/setup.nsi share/qt/Info.plist test/config.ini])
 AC_CONFIG_FILES([contrib/devtools/split-debug.sh],[chmod +x contrib/devtools/split-debug.sh])
 AM_COND_IF([HAVE_DOXYGEN], [AC_CONFIG_FILES([doc/Doxyfile])])
diff --git a/src/httpserver.cpp b/src/httpserver.cpp
index 65c2967cfa..2847dd75b4 100644
--- a/src/httpserver.cpp
+++ b/src/httpserver.cpp
@@ -2,6 +2,10 @@
 // Distributed under the MIT software license, see the accompanying
 // file COPYING or http://www.opensource.org/licenses/mit-license.php.
 
+#if defined(HAVE_CONFIG_H)
+#include <config/bitcoin-config.h>
+#endif
+
 #include <httpserver.h>
 
 #include <chainparamsbase.h>
@@ -597,7 +601,13 @@ CService HTTPRequest::GetPeer() const
         // evhttp retains ownership over returned address string
         const char* address = "";
         uint16_t port = 0;
+
+#ifdef HAVE_EVHTTP_CONNECTION_GET_PEER_CONST_CHAR
+        evhttp_connection_get_peer(con, &address, &port);
+#else
         evhttp_connection_get_peer(con, (char**)&address, &port);
+#endif // HAVE_EVHTTP_CONNECTION_GET_PEER_CONST_CHAR
+
         peer = LookupNumeric(address, port);
     }
     return peer;
```

[the remark this answers is not in the document]
Comment (PastaPastaPasta) [reply configure.ac:1501]: I managed to have this commit twice; second (partial) one dropped

File: test/functional/test_framework/authproxy.py [hunk 3]
```diff
@@ -113,10 +113,8 @@ def _request(self, method, path, postdata):
             self.__conn.request(method, path, postdata, headers)
             return self._get_response()
         except OSError as e:
-            retry = (
-                '[WinError 10053] An established connection was aborted by the software in your host machine' in str(e))
             # Workaround for a bug on macOS. See https://bugs.python.org/issue33450
-            retry = retry or ('[Errno 41] Protocol wrong type for socket' in str(e))
+            retry = '[Errno 41] Protocol wrong type for socket' in str(e)
```
[the remark this answers is not in the document]
Comment (PastaPastaPasta) [reply test/functional/test_framework/authproxy.py:117]: dropped

File: src/validation.h (hunk 1 above)
Comment (PastaPastaPasta) [reply src/validation.h:292]: resolved

**Review by knst — APPROVED** [review]
utACK

**Review by UdjinM6 — APPROVED** [review]
utACK

**Pull request merged**